### PR TITLE
Open Earn - Simulate main section

### PR DESCRIPTION
**Pull request by jakubswierczek**
# [Open Earn - Simulate main section](https://app.shortcut.com/oazo-apps/story/3420/simulate-main-section)
  
## Changes 👷‍♀️
* New new on open earn page  
## How to test 🧪
* Open earn and try to navigate between tabs,
* Open manage vault and again try to navigate between tabs,
* Try the same with borrow or multiply

### CODE REVIEW

File: components/DetailsSection.tsx [hunk 1]
```diff
@@ -20,43 +21,8 @@ export function DetailsSection({ title, badge, buttons, content, footer }: Detai
...
-              }}
-            >
-              {buttons?.map((button, i) => (
-                <ExpandableButton button={button} key={i} />
-              ))}
-            </Flex>
-          )}
-        </Flex>
-      )}
+      {title && isString(title) && <TitleWrapper title={title} badge={badge} buttons={buttons} />}
```
Comment (paszkowskiDamian) [inline components/DetailsSection.tsx:24]: I think we do not need lodash function to test if something is a string. `typeof title === 'string'`

File: components/DetailsSectionContentTable.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+import { oasisBaseTheme } from 'theme'
+import { Box, Card, Flex, Grid, Text } from 'theme-ui'
+import { useMediaQuery } from 'usehooks-ts'
+
+import { ChevronUpDown } from './ChevronUpDown'
+import { SelectComponents } from 'react-select/src/components'
+
+interface ContentTableProps {
+  headers: [string, string, string]
+  rows: [string, string, string][]
```
Comment (paszkowskiDamian) [inline components/DetailsSectionContentTable.tsx:12]: Is there a reason behind constraining the table to have just three columns?

File: components/vault/GeneralManageTabBar.tsx [hunk 3]
```diff
@@ -53,7 +53,8 @@ export function GeneralManageTabBar({
   useEffect(() => {
     const uiChanges$ = uiChanges.subscribe<TabChange>(TAB_CHANGE_SUBJECT)
     const subscription = uiChanges$.subscribe((value) => {
-      setMode(() => value.currentMode)
+      console.log(`TabChange: ${value}`)
```
Comment (paszkowskiDamian) [inline components/vault/GeneralManageTabBar.tsx:56]: I guess you forgot to remove it 😉

File: components/vault/detailsSection/ContentFooterItemsEarnSimulate.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  breakeven,
+  entryFees,
+  apy,
+}: ContentFooterItemsEarnSimulateProps) {
+  const { t } = useTranslation()
+
+  const formatted = {
+    breakeven: breakeven.gt(zero) ? breakeven.toFixed(2) : '-',
+    entryFees: entryFees.gt(zero) ? `${entryFees.toFixed(2)} ${token}` : '-',
+    apy: apy.toFixed(2),
```
Comment (paszkowskiDamian) [inline components/vault/detailsSection/ContentFooterItemsEarnSimulate.tsx:25]: we have functions for formatting some values such as `formatPercent`.

File: components/vault/detailsSection/ContentFooterItemsEarnSimulate.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+export function ContentFooterItemsEarnSimulate({
+  token,
+  breakeven,
+  entryFees,
+  apy,
+}: ContentFooterItemsEarnSimulateProps) {
+  const { t } = useTranslation()
+
+  const formatted = {
+    breakeven: breakeven.gt(zero) ? breakeven.toFixed(2) : '-',
```
Comment (paszkowskiDamian) [inline components/vault/detailsSection/ContentFooterItemsEarnSimulate.tsx:23]: Isn't `breakeven` denominated in days or something?

File: features/earn/guni/open/containers/GuniOpenMultiplyVaultDetails.tsx [hunk 6]
```diff
@@ -1,64 +1,169 @@
...
+  const oazoFee = daiAmountToSwapForUsdc.times(OAZO_LOWER_FEE)
+
+  if (actualGasCostInDai) {
+    const daiGasFee = actualGasCostInDai
+    return daiGasFee.plus(oazoFee)
+  }
+
+  return oazoFee.plus(
+    examplePosition.gasUsed.times(amountFromWei(gasPrice).times(ETHUsd).div(DAIUsd)),
+  )
```
Comment (sennett) [inline features/earn/guni/open/containers/GuniOpenMultiplyVaultDetails.tsx:45]: Do we need this?  Can't we run the gas estimation routine on the 150,000 placeholder value without this backup calc?

File: components/DetailsSection.tsx (hunk 1 above)
Comment (jakubswierczek) [reply components/DetailsSection.tsx:24]: resolved

File: components/DetailsSection.tsx [hunk 7]
```diff
@@ -78,3 +44,49 @@ export function DetailsSection({ title, badge, buttons, content, footer }: Detai
...
+        {badge !== undefined && <VaultTabTag isEnabled={badge} />}
+      </Flex>
+      {buttons && (
+        <Flex
+          sx={{
+            mt: [2, null, 0],
+          }}
+        >
+          {buttons?.map((button, i) => (
+            <ExpandableButton button={button} key={i} />
```
[the remark this answers is not in the document]
Comment (jakubswierczek) [reply components/DetailsSection.tsx:86]: done

File: components/vault/EarnVaultSimulateTabBar.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  Simulate = 'simulate',
+  FAQ = 'faq',
+}
+
+interface EarnVaultSimulateTabBarProps {
+  detailsAndForm: JSX.Element
+  faq?: ReactElement
+}
+
+export function EarnVaultSimulateTabBar({
```
[the remark this answers is not in the document]
Comment (jakubswierczek) [reply components/vault/EarnVaultSimulateTabBar.tsx:21]: I will remove it.

File: components/DetailsSectionContentTable.tsx (hunk 2 above)
Comment (jakubswierczek) [reply components/DetailsSectionContentTable.tsx:12]: change to `string[][]`

File: features/earn/guni/open/containers/GuniOpenMultiplyVaultDetails.tsx (hunk 6 above)
Comment (jakubswierczek) [reply features/earn/guni/open/containers/GuniOpenMultiplyVaultDetails.tsx:53]: I'm not sure. Can we estimate gas when I don't have a proxy?

**Review by sennett — APPROVED** [review]

**Pull request merged**